### ISSUE TITLE
[evidence] Upload file button - must have sufficient color contrast

### ISSUE THREAD

**Issue by khalidelaggan** [cds-snc/report-a-cybercrime#1177]
## Summary

> Please ensures that the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds

## Design Detail

<img width="177" alt="Screen Shot 2020-01-19 at 7 23 23 PM" src="https://user-images.githubusercontent.com/2603117/72691183-67ba1880-3af1-11ea-94f4-62ae1956c7f6.png">

## Recommended Solution

To solve this violation, you need to fix the following:
Element has insufficient color contrast of 4.02 (foreground color: #ffffff, background color: #3182ce, font size: 13.5pt (18px), font weight: normal). Expected contrast ratio of 4.5:1

<img width="515" alt="Screen Shot 2020-01-19 at 7 27 25 PM" src="https://user-images.githubusercontent.com/2603117/72691211-bf588400-3af1-11ea-9aa3-96b6a4974220.png">

## Checker result 
<img width="1084" alt="Screen Shot 2020-01-19 at 7 37 36 PM" src="https://user-images.githubusercontent.com/2603117/72691416-37737980-3af3-11ea-8825-4f67d8e429a8.png">

**Comment by khalidelaggan** [comment]
I'm assuming Figma has a built in contrast checker, however I'm adding here my favourite tool 

https://whocanuse.com/

**Comment by amazingphilippe** [comment]
I think this is a classic case of the [white-on-blue wcag fail](https://github.com/w3c/wcag/issues/695). I would suggest staying away from this combination of colors. 

We could go towards a neutral palette. We want to avoid greens and reds because they mean something, and we want to avoid failing color contrasts like white-on-blue. A neutral palette or a darker blue seems like a simple fix for now.

**Comment by Myndex** [comment]
> I think this is a classic case of the [white-on-blue wcag fail](https://github.com/w3c/wcag/issues/695). I would suggest staying away from this combination of colors.

HI @amazingphilippe  I happened to come over here as I saw you linked to the issue I created last year regarding the WCAG contrast math problem. I am the principal researcher for the new visual contrast methods and standards, in the process of fixing this problem.

If you have a requirement to meet WCAG 2.x, then you can get the same perceptual look with a minor change: make the color **#3177CE** 

#### You'll notice that I only lowered the GREEN very slightly!

This is because that VAST majority of luminance comes from green. Blue contributes almost none. As a result you'll see that #3177CE looks almost the same as #3182CE, except that #3177CE BG and #FFF text meets WCAG 4.5:1 using the WCAG math. TIP: if you have a color pair that you like but fails the current WCAG but is close, make a change to the green channel to improve contrast.

## The Future
FWIW, the reverse contrast issue such as this white on blue is an example of the WCAG false fails. The new standard will not be published for a couple years, and if you have a legal requirement to follow the current 2.x standard, **then disregard what I am about to say.** What follows is NOT normative and not yet a part of any official standard.

The nature of human visual perception is such that blue is fine as a design element, provided it is the darkest of a given pair of colors. The fact that WCAG fails it is due to some incorrect math, not an actual accessibility issue. In fact I have some ongoing experiments that show that white on a medium to dark blue background can actually be an accessibility _HELPER_ for some forms of impairment, particularly older eyes that suffer from glare/scatter.

But also realize that helping one impairment may be negative for another. However, the only color vision deficiency that can have a problem with white text on blue  BG is the very rare blue cone monochromat (something like 1 in a million), and they need assistive technologies (monochrome monitors, reduced luminance environments, etc.)

I have a live contrast checker tool that demonstrates the work in progress. It is preliminary, and not the standard as yet, but feel free to bash around with it — you'll see there are fewer false fails, but more importantly, fewer false _PASSES_ which is the more substantial problem we are working to solve. 

Work In Progress SAPC Contrast Tool: https://www.myndex.com/SAPC/

Along with this, I also have a CVD simulator based on the Brettel model (generally considered an accurate representation of the most common color vision deficiencies) which is here: https://www.myndex.com/CVD/

Please feel free to ask me any questions.


Regards,

Andrew Somers
_Color Science Researcher_

**Comment by khalidelaggan** [comment]
Thank you so much Andrew for your insight on this, greatly appreciated!

**Comment by khalidelaggan** [comment]
@amazingphilippe I think what makes this issue slightly complex is that we are currently not using a true white background and it's instead we are using #F7FAFC
<img width="241" alt="Screen Shot 2020-01-22 at 9 12 54 AM" src="https://user-images.githubusercontent.com/2603117/72901233-63f7e300-3cf7-11ea-8d36-349b6b46b188.png">

Assigning the background color to #FFFFFF and the links color to #3177CE  as Andrew suggested should meet the expectations of both worlds :) unless it's not following our design standard.
